docs/IE_DG/network_state_intro.md

```diff
@@ -0,0 +1,193 @@
+Introduction to OpenVINO state API {#openvino_docs_IE_DG_network_state_intro}
```

Comment:
we need to add this document to `ie_docs.xml` layout file. Otherwise, it's not a part of documentation.